sql/core/src/main/scala/org/apache/spark/sql/execution/dynamicpruning/PartitionPruning.scala

```diff
@@ -85,18 +85,19 @@ object PartitionPruning extends Rule[LogicalPlan] with PredicateHelper {
       filteringKey: Expression,
       filteringPlan: LogicalPlan,
       joinKeys: Seq[Expression],
-      hasBenefit: Boolean): LogicalPlan = {
+      partScan: Option[LogicalRelation]): LogicalPlan = {
```

Comment:
Can you do a follow-up PR changing this to `LogicalRelation` instead of `Option[LogicalRelation]`, @monkeyboy123 ?

Reply:
ok,i will do it right now

Reply:
@maryannxue  done! thanks for review